src/main/java/meep/parser/Parser.java

```diff
@@ -0,0 +1,179 @@
+package meep.parser;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import meep.task.Deadline;
+import meep.task.Event;
+import meep.task.TaskList;
+import meep.task.Todo;
+import meep.ui.Ui;
+
+/**
+ * The {@code Command} enum represents the various commands that can be issued
+ * by the user to interact with the task management system.
+ */
+enum Command {
+    BYE,
+    MARK,
+    UNMARK,
+    DELETE,
+    DEADLINE,
+    EVENT,
+    TODO,
+    LIST,
+    FIND
+}
+
+/**
+ * The {@code Parser} class processes user input, identifies commands, and
+ * executes the corresponding actions on a {@code TaskList}. It interacts with
+ * the user interface and handles various commands such as adding tasks,
+ * marking tasks as done or undone, and deleting tasks.
+ */
+public class Parser {
+    private final Ui ui;
+
+    /**
+     * Constructs a {@code Parser} with a specified user interface.
+     */
+    public Parser() {
+        this.ui = new Ui();
+    }
+
+    /**
+     * Returns the appropriate suffix (st, nd, rd, th) for a given day of the month.
+     *
+     * @param day The day of the month.
+     * @return The suffix for the day.
+     */
+    private String getDayOfMonthSuffix(int day) {
+        if (day >= 11 && day <= 13) {
+            return "th";
+        }
+        return switch (day % 10) {
+        case 1 -> "st";
+        case 2 -> "nd";
+        case 3 -> "rd";
+        default -> "th";
+        };
```

Comment:
Perhaps there is a better way to use switch in the coding standard?

Reply:
Thanks for the feedback. The IDE suggested using this method instead of the traditional notation, so I'm not sure if this notation complies with the coding standards.